External/Plugins/ProjectManager/Actions/FileActions.cs

```diff
@@ -446,33 +446,31 @@ public bool Rename(string oldPath, string newName)
                 if (isDirectory)
                 {
                     // this is required for renaming directories, don't ask me why
-                    string oldPathFixed = (oldPath.EndsWith('\\')) ? oldPath : oldPath + "\\";
-                    string newPathFixed = (newPath.EndsWith('\\')) ? newPath : newPath + "\\";
-                    if (oldPathFixed.Equals(newPathFixed, StringComparison.OrdinalIgnoreCase))
+                    if (oldPath.EndsWith('\\')) oldPath = oldPath.Substring(0, oldPath.Length - 1);
+                    if (newPath.EndsWith('\\')) newPath = newPath.Substring(0, newPath.Length - 1);
+                    if (oldPath.Equals(newPath, StringComparison.OrdinalIgnoreCase))
                     {
                         // name casing changed
-                        string tmpPath = newPathFixed.Substring(0, newPathFixed.Length - 1) + "$renaming$\\";
-                        Directory.Move(oldPathFixed, tmpPath);
-                        oldPathFixed = tmpPath;
+                        string tmpPath = newPath + "$renaming$";
+                        Directory.Move(oldPath, tmpPath);
+                        Directory.Move(tmpPath, newPath);
                     }
-                    if (FileHelper.ConfirmOverwrite(newPath))
+                    else if (FileHelper.ConfirmOverwrite(newPath))
                     {
-                        FileHelper.ForceMoveDirectory(oldPathFixed, newPathFixed);
+                        FileHelper.ForceMoveDirectory(oldPath, newPath);
                     }
                     else return false;
                 }
                 else
                 {
-                    string oldName = Path.GetFileName(oldPath);
-                    if (oldName.Equals(newName, StringComparison.OrdinalIgnoreCase))
+                    if (oldPath.Equals(newPath, StringComparison.OrdinalIgnoreCase))
                     {
                         // name casing changed
                         string tmpPath = newPath + "$renaming$";
                         File.Move(oldPath, tmpPath);
-                        oldPath = tmpPath;
+                        File.Move(tmpPath, newPath);
```

Comment:
I see a little gotcha here. Since you are calling directly File.Move, in case the file is the main target one, it doesn't get updated in the project. Of course, in Windows it's OK, but it wouldn't be in case sensitive scenarios.